### PR TITLE
[bitnami/thanos] fix sharded store dnsConfig error

### PR DESCRIPTION
**Pull request by antoinedeschenes**
### Description of the change

Helm release fails with the following error when using a sharded store:
```
Error: exit status 1: Error: template: thanos/charts/thanos/templates/storegateway/statefulset-sharded.yaml:75:20: executing "thanos/charts/thanos/templates/storegateway/statefulset-sharded.yaml" at <.Values.storegateway.dnsConfig>: can't evaluate field Values in type int
```

### Benefits

Fix sharded store deployments

### Applicable issues

- fixes bug introduced by #15694 

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Comment by aghassabian** [comment]
@migruiz4 Merging this one would unblock us. Any ETA?

File: bitnami/thanos/Chart.yaml [hunk 1]
```diff
@@ -28,4 +28,4 @@ name: thanos
 sources:
   - https://github.com/bitnami/containers/tree/main/bitnami/thanos
   - https://thanos.io
-version: 12.4.0
```
Comment (migruiz4) [inline bitnami/thanos/Chart.yaml:31]: Version 12.4.1 has already been released, could you please bump the chart version to `12.4.2`?

https://github.com/bitnami/charts/blob/main/bitnami/thanos/Chart.yaml

**Review by migruiz4 — CHANGES_REQUESTED** [review]
Hi @antoinedeschene,

Thank you very much for your contribution! The changes look good to me, but could you please bump the chart version again?

File: bitnami/thanos/Chart.yaml (hunk 1 above)
Comment (antoinedeschenes) [reply bitnami/thanos/Chart.yaml]: @migruiz4 Thanks, updated to 12.4.2

**Comment by antoinedeschenes** [comment]
Closing, duplicated by https://github.com/bitnami/charts/pull/15827.

**Pull request closed without merging**